nixos/modules/services/audio/spotifyd.nix

```diff
@@ -0,0 +1,47 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.spotifyd;
+  spotifydConf = pkgs.writeText "spotifyd.conf" cfg.config;
+in
+{
+  options = {
+    services.spotifyd = {
+      enable = mkEnableOption "spotifyd, a Spotify playing daemon";
+
+      config = mkOption {
+        default = "";
+        type = types.lines;
+        description = ''
+          Configuration for Spotifyd. For syntax and directives, see
+          https://github.com/Spotifyd/spotifyd#Configuration.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.spotifyd = {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" "sound.target" ];
```

Comment:
network-online.target?

Reply:
Is there any particular reason we would want `network-online.target` instead of `network.target`?

As I did not know the difference between `network.target` and `network-online.target`, I did a quick search and found https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/ where they say this:


> It is strongly recommended not to pull in this target [network-online.target] too liberally: for example network server software should generally not pull this in (since server software generally is happy to accept local connections even before any routable network interface is up), its primary purpose is network client software that cannot operate without network.

Reply:
I was not aware of the actual difference either, and just wanted you to consider it as I've seen we've been adding `network-online.target` in various places lately.

Reply:
Alright, so let us stick with `network.target`, at least until we have a good reason to switch to `network-online.target`?

Reply:
I'm pretty sure `network-online.target` is actually what we need here, spotify can't operate without network, which is exactly what the description mentions as its primary purpose.

Reply:
Ok, fixed!